salt/utils/data.py

```diff
@@ -101,41 +119,46 @@ def decode(data, encoding=None, errors='strict', keep=False,
 
 
 def decode_dict(data, encoding=None, errors='strict', keep=False,
-                preserve_dict_class=False, preserve_tuples=False):
+                normalize=False, preserve_dict_class=False,
+                preserve_tuples=False):
     '''
     Decode all string values to Unicode
     '''
     # Make sure we preserve OrderedDicts
     rv = data.__class__() if preserve_dict_class else {}
     for key, value in six.iteritems(data):
         if isinstance(key, tuple):
-            key = decode_tuple(key, encoding, errors, keep, preserve_dict_class) \
+            key = decode_tuple(key, encoding, errors, keep, normalize,
+                               preserve_dict_class) \
                 if preserve_tuples \
-                else decode_list(key, encoding, errors, keep,
+                else decode_list(key, encoding, errors, keep, normalize,
                                  preserve_dict_class, preserve_tuples)
         else:
             try:
-                key = salt.utils.stringutils.to_unicode(key, encoding, errors)
+                key = salt.utils.stringutils.to_unicode(
+                    key, encoding, errors, normalize)
             except TypeError:
                 pass
```

Comment:
I would add here a comment that `pass` is only technical here as it is indicates that `to_unicode` did not processed correct type.